calico_kubernetes/calico_kubernetes.py

```diff
@@ -147,6 +165,117 @@ def _configure_interface(self):
         print('Finished configuring network interface')
         return ep
 
+    def _container_add(self, container_name, workload_id, pid, ip, interface,
+                       hostname, orchestrator_id):
+        """
+        Add a container (on this host) to Calico networking with the given IP.
+
+        :param container_name: The name or ID of the container.
+        :param ip: An IPAddress object with the desired IP to assign.
+        :param interface: The name of the interface in the container.
+        """
+        # Check if the container already exists. If it does, exit.
+        try:
```

Comment:
Eh, not sure that makes a significant different.  It saves a few lines perhaps, but the exception handling is fine I think.